tests/unit/FormDatePicker.spec.js

```diff
@@ -0,0 +1,157 @@
+import { shallowMount } from '@vue/test-utils'
+import FormDatePicker from '../../src/components/FormDatePicker.vue';
+
+describe('FormDatePicker', () => {
+  let wrapper;
+  beforeEach(() => {
+    wrapper = shallowMount(FormDatePicker, {
+      propsData: {
+        dataFormat: 'date'
+      }
+    });
+  });
+
+  it('can create the component', () => {
+    expect(wrapper.html()).toContain('date-picker-stub');
+
+    const value = wrapper.find('date-picker-stub').vm.value;
+    const dateDisplayed = new Date(value).setUTCMilliseconds(0);
+    const today = new Date().setUTCMilliseconds(0);
+
+    expect(dateDisplayed).toBe(today);
+  });
+
+
+  it('should render all configured props', () => {
+    const nameText = 'birthdate';
+    const labelText = 'Enter Your Birthday';
+    const placeholderText = 'MM/DD/YYYY';
+    const helperText = 'This is some text';
+    const errorText = 'This field has an error';
+    const readOnly = true;
+
+    wrapper.setProps({
+      name: nameText,
+      label: labelText,
+      placeholder: placeholderText,
+      helper: helperText,
+      error: errorText,
+      disabled: readOnly,
+      validation: 'required'
+    });
+    expect(wrapper.html()).toContain(label);
+    expect(wrapper.vm.placeholder).toBe(placeholder);
+    expect(wrapper.find('.invalid-feedback').text()).toContain(helperText);
+    expect(wrapper.find('.invalid-feedback').text()).toContain(errorText);
+    expect(wrapper.vm.disabled).toBe(true);
+    expect(wrapper.find('date-picker-stub').vm.props.config.format).toBe('MM/DD/YYYY');
+
+    expect(wrapper.name()).toBe(nameText);
+    expect(wrapper.vm.dataFormat).toBe(dataType);
+
+    wrapper.setProps({
+      dataFormat: 'datetime'
+    });
+    expect(wrapper.find('date-picker-stub').vm.props.config.format).toBe('MM/DD/YYYY h:mm A');
+  });
+
+  it('updates and emits values', async () => {
+    let date = new Date('10/20/2020').toISOString();
+    // Should emit on mount
+    expect(wrapper.emitted().input).toBeTruthy();
+    
+    
+    wrapper.setProps({value: date});
+    
+    expect(wrapper.emitted().input.length).toBe(2);
+    expect(wrapper.emitted().input[1]).toEqual(['10/20/2020']);
+    expect(wrapper.vm.value).toBe(date);
+
+    date = new Date('10/20/2020 10:00 AM').toISOString();
+    wrapper.setProps({
+      dataFormat: 'datetime',
+      value: date
+    });
+    expect(wrapper.emitted().input.length).toBe(3);
+    expect(wrapper.emitted().input[2]).toBe('10/20/20 10:00 AM');
+    expect(wrapper.vm.value).toBe(date);
+  });
+
+  it('runs validation for date', () => {
+    const requiredText = 'The birthdate field is required.';
+    const invalidText = 'The birthdate is not a valid format.';
+
+    wrapper.setProps({
+      name: 'birthdate',
+      validation: 'required'
+    });
+    
+    // Test  validation by removing values
```

Comment:
Keep tests small and focused. If you find yourself adding comments like these, consider breaking it out into a separate "test validation by removing values" test.